towerlib/towerlib.py

```diff
@@ -1908,7 +1908,14 @@ def create_job_template(self,  # pylint: disable=too-many-arguments, too-many-lo
         response = self.session.post(url, json=payload)
         if not response.ok:
             self._logger.error('Error creating job template, response was: "%s"', response.text)
-        return JobTemplate(self, response.json()) if response.ok else None
+            return None
+        job_template = JobTemplate(self, response.json())
+        if credential_id:
+            if not job_template.add_credential_by_id(credential_id):
+                self._logger.error('Error adding credentials, reverting the creation of the job template')
+                job_template.delete()
```

Comment:
Since we are deleting here we should also check the output of that job so if for any reason it goes wrong it does not happen silently. Probably should raise an exception if it does not work?